.github/workflows/ci.yml

```diff
@@ -33,18 +39,28 @@ jobs:
           node-version: "12"
 
       - id: "haskell"
+        name: "(Non-Linux only) Install Haskell"
+        if: "${{ runner.os != 'Linux' }}"
         uses: "haskell/actions/setup@v1"
         with:
           enable-stack: true
           # If upgrading Stack, also upgrade it in the lint job below
-          stack-version: "2.7.1"
+          stack-version: "${{ env.STACK_VERSION }}"
           stack-no-global: true
 
+      - name: "(Linux only) Check Stack version and ensure a (dummy) accessible local repository"
+        if: "${{ runner.os == 'Linux' }}"
+        run: |
+          [ "$(stack --numeric-version)" = "$STACK_VERSION" ]
+          echo 'allow-different-user: true' >>/root/.stack/config.yaml
```

Comment:
Might be simpler to `chown root:root .`, unless we expect that to cause some sort of problem? Everything else created in the working directory will be owned by root.